Summatives/m3-summative/LINQ/Program.cs

```diff
@@ -0,0 +1,732 @@
+        /// Print the list of products ordered first by category, then by unit price, from highest to lowest.
+        /// </summary>
+        static void PrintAllProductsSortByCategoryandUnitPrice()
+        {
+            List<Product> products = DataLoader.LoadProducts();
+            var productssortbyCategoryandUnitPrice = products.OrderBy(p => p.Category).ThenByDescending(p => p.UnitPrice).ToList();
+            PrintProductInformation(productssortbyCategoryandUnitPrice);
+        }
+
+        /// <summary>
+        /// Print NumbersB in reverse order
+        /// </summary>
+        static void PrintNumbersBinreverse()
+        {
+            var reversenumbers = DataLoader.NumbersB.Reverse();
+
+            string line = "{0,-19}";
+            Console.WriteLine(line, "NumbersB in Reverse");
+            Console.WriteLine("===================");
+
+            foreach (var number in reversenumbers)
+            {
+                Console.WriteLine(line, number);
+            }
+        }
+
+        /// <summary>
+        /// Group products by category, then print each category name and its products
+        /// ex:
+        /// 
+        /// Beverages
+        /// Tea
+        /// Coffee
+        /// 
+        /// Sandwiches
+        /// Turkey
+        /// Ham
+        /// </summary>
+        static void PrintCategoryandProducts()
+        {
+            List<Product> products = DataLoader.LoadProducts();
+            var categoryproducts = from product in products
+                                   orderby product.Category, product.ProductName
+                                   group product by product.Category;
+
+            string line = "{0,-50}";
+            Console.WriteLine(line, "Category and Product Names");
+            Console.WriteLine("==============================================");
+
+            foreach (var category in categoryproducts)
+            {
+                Console.WriteLine(line, category.Key);
+                foreach (var product in category)
+                {
+                    Console.WriteLine("\t{0}", product.ProductName);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Print all Customers with their orders by Year then Month
+        /// ex:
+        /// 
+        /// Joe's Diner
+        /// 2015
+        ///     1 -  $500.00
+        ///     3 -  $750.00
+        /// 2016
+        ///     2 - $1000.00
+        /// </summary>
+        static void PrintCustomersOrderYearMonth()
+        {
+            var customers = DataLoader.LoadCustomers();
+            var customerssortbyorder = from customer in customers
+                                       where customer.Orders.Any() is true
+                                       select new
+                                       {
+                                           CompanyName = customer.CompanyName,
+                                           Orders = customer.Orders.OrderBy(order => order.OrderDate)
```

Comment:
You'll want to nest 2-3 LINQ statements that will group the orders by Year, then by Month